src/components/webview/MainWebviewProvider.ts

```diff
@@ -480,22 +480,34 @@ export class IntuitaProvider implements WebviewViewProvider {
 			if (message.value.command === 'vscode.diff') {
 				const args = message.value.arguments;
 
-				if (!args) {
-					return;
+				if (!args?.[0]?.path || !args?.[1]?.path) {
+					throw new Error(
+						'Expected args[0] and args[1] to be resource Uri',
+					);
 				}
 
-				const left = Uri.parse(args[0].path);
-				const right = Uri.parse(args[1].path);
+				const leftUri = Uri.parse(args[0].path);
+				const rightUri = Uri.parse(args[1].path);
 				const title = args[2];
 
 				commands.executeCommand(
 					message.value.command,
-					left,
-					right,
+					leftUri,
+					rightUri,
 					title,
 				);
 			}
 
+			if (message.value.command === 'vscode.open') {
+				const args = message.value.arguments;
+				if (!args?.[0]?.path) {
```

Comment:
does `args[0] instanceof Uri` work?

Reply:
no, it will be false because we pass it from webview. I think it serializes it to string and then back when we use postMessage. (i will double check that)